tests/testthat/test-observer.R

```diff
@@ -0,0 +1,47 @@
+test_that("can finalise observations automatically where similar", {
+  expect_equal(
+    observer_finalise_auto(list(list(a = 1:5), list(a = 6:10))),
+    list(a = cbind(1:5, 6:10)))
+  expect_equal(
+    observer_finalise_auto(list(list(a = 1:5, b = 1:3),
+                                list(a = 6:10, b = 4:6))),
+    list(a = cbind(1:5, 6:10),
+         b = cbind(1:3, 4:6)))
+})
+
+
+test_that("don't combine things that can't be stuck together", {
+  expect_equal(
+    observer_finalise_auto(list(list(a = 1:5, b = 1:3), list(a = 6:10))),
+    list(list(a = 1:5, b = 1:3), list(a = 6:10)))
+  expect_equal(
+    observer_finalise_auto(list(list(a = 1:5, a = 1), list(a = 6:10, a = 1))),
+    list(list(a = 1:5, a = 1), list(a = 6:10, a = 1)))
+  expect_equal(
+    observer_finalise_auto(list(list(1:5, 1), list(6:10, 1))),
+    list(list(1:5, 1), list(6:10, 1)))
```

Comment:
why have we chosen to not output `list(cbind(1:5, 6:10), cbind(1, 1))`? is it because this could be as deep as the user wants and we would have to recursively keep checking dimensions?